docs/templates/base.html

```diff
@@ -0,0 +1,144 @@
+  let currentLang = null;
+  codeSelector.childNodes.forEach((element) => {
+    let toAdd = null;
+
+    if (element.nodeType === Node.COMMENT_NODE) {
+      // if its a comment like <!-- go -->, then set the currentLang to go and add it to langs
+      // end will cancel the currentLang, so you can continue to write common documentation.
+      const lang = element.data.trim();
+      if (lang === "end") {
+        currentLang = null;
+      } else {
+        currentLang = lang;
+        toAdd = { lang, element };
+      }
+    } else if (element.tagName === "PRE") {
+      // pre with data-lang attribute
+      const lang = element.getAttribute("data-lang");
+      // code blocks reset the currentLang.
+      currentLang = null;
+      toAdd = { lang, element };
+    } else if (currentLang) {
+      // if we are in a currentLang, then add the element to it.
+      toAdd = { lang: currentLang, element };
+    }
+
+    if (!toAdd) {
+      return;
+    }
+
+    langs.push(toAdd);
+    if (tabs.indexOf(toAdd.lang) === -1) {
+      tabs.push(toAdd.lang);
+    }
+  });
+
+  // Let us put this in the codeSelector element, so that we later check if the new selected language is in the list first.
+  codeSelector.langs = langs;
+
+  let saved = localStorage.getItem("code-lang");
+  let selected = tabs[0];
+  if (saved) {
+    // Only select the saved language if it's in the list.
+    found = tabs.find((lang) => lang === saved);
+    if (found) {
+      selected = found;
+    }
+  }
+
+  for (const lang of tabs) {
+    const isSelected = selected === lang;
+
+    const li = document.createElement("li");
+    li.classList.add("nav-item");
+
+    const a = document.createElement("a");
+    a.classList.add("nav-link");
+    if (isSelected) {
+      a.classList.add("active");
+      a.setAttribute("aria-current", "page");
+    }
+
+    a.href = "#";
+    a.textContent = capitalize(lang);
+    a.lang = lang;
+
+    a.addEventListener("click", (e) => {
+      e.preventDefault();
+      changeLanguage(lang);
+    });
+
+    li.appendChild(a);
+    nav.appendChild(li);
+  }
+
+  for (const { lang, element } of langs) {
+    const isSelected = selected === lang;
+    if (element.classList) {
+      if (!isSelected) {
+        element.classList.add("d-none");
+      }
+    }
+  }
+}
+
+function capitalize(str) {
+  return str[0].toUpperCase() + str.slice(1);
+}
+
+function changeLanguage(lang) {
+  localStorage.setItem("code-lang", lang);
+
+  document.querySelectorAll(".code-selector").forEach((codeSelector) => {
+    const langs = codeSelector.langs;
+    if (!langs) {
+      console.error("Missing langs property on codeSelector", codeSelector);
+      return;
+    }
+
+    const selected = langs.find((l) => l.lang === lang);
+    if (!selected) {
+      // This tab group doesn't have the selected language--all good.
+      return;
+    }
+
+    // Show/hide each element within the codeSelector
+    langs.forEach((l) => {
+      if (l.element.classList) {
```

Comment:
Huh, when is l.element.classList undefined? All elements should have classes, or at least have the interface for adding them...

Reply:
When scanning I did see `\n`'s between `<p>`'s so I assumed it was those. Actually that was the case for styles, and copied the logic over to classList without checking.

Reply:
Interesting. Just checked it again, and same applies to `classList`. `#text` newline elements and also comments had no classLists, at least in Chrome.